crates/eips/src/eip4844/mod.rs

```diff
@@ -104,6 +104,85 @@ where
     Ok(blob)
 }
 
+/// A heap allocated blob that serializes as 0x-prefixed hex string
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    alloy_rlp::RlpEncodableWrapper,
+    alloy_rlp::RlpDecodableWrapper,
+)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
+#[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
+pub struct HeapBlob(Bytes);
+
+impl HeapBlob {
+    /// Create a new heap blob from a byte slice.
+    pub fn new(blob: &[u8]) -> Result<Self, InvalidBlobLength> {
+        if blob.len() != BYTES_PER_BLOB {
+            return Err(InvalidBlobLength(blob.len()));
+        }
+
+        unsafe { Ok(Self::new_unchecked(Bytes::copy_from_slice(blob))) }
+    }
+
+    /// Create a new heap blob from a byte slice without checking the length.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure that the length of `blob` is [`BYTES_PER_BLOB`].
+    pub unsafe fn new_unchecked(blob: Bytes) -> Self {
+        Self(blob)
+    }
```

Comment:
this is safe

but this function shouldn't exist because this is a massive footgun